dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/JsonResponseFile.java

```diff
@@ -72,24 +72,24 @@ private static ObjectMapper newObjectMapper() {
   }
 
   @Nonnull
-  private static List<? extends String> to_arguments(@Nonnull File file) throws IOException {
```

Comment:
I would suggest to rewrite `List<? extends T>` to `ImmutableList<T>`  to maintain the guarantee that the original code did provide (i.e. you can't `add` to the list).

Reply:
Though adding to such a list is tricky, I think there's no _guarantee_ that it won't happen. For example,
```
<T> void fun(List<T> x) {
  x.add(x.get(0));
}

...

fun(returned list);
```
successfully modifies a (nonempty) `List<? extends String>` object

Reply:
> I would suggest to rewrite `List<? extends T>` to `ImmutableList<T>` to maintain the guarantee that the original code did provide (i.e. you can't `add` to the list).

Though adding to such a list is tricky, I think there's no _guarantee_ that it won't happen. For example,

```
<T> void fun(List<T> x) {
  x.add(x.get(0));
}

...

fun(returned list);
```
successfully modifies a (nonempty) `List<? extends String>` object

Reply:
@paolomorandini I didn't notice this guarantee in the original code. However, the return type should be an immutable collection, whenever possible. If this was the intention behind how the code was written before, then I agree with you that rewriting to `ImmutableList` would maintain this intention (either of immutability or of partial immutability, such as that `add` cannot be called (even though @misolt already provided a counter-example)).

I've applied the change.